includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -516,6 +516,13 @@ public function should_show_payment_request_button() {
 			return false;
 		}
 
+		// Make sure to prevent displaying products when price is not set.
+		$product_price = $this->get_product_price( $this->get_product() );
```

Comment:
It's not safe to call this method here. It will cause troubles on cart and checkout pages. I see there is even better method above on line 508. 

`if ( $this->is_product() && ! $this->is_product_supported() )`

So if we move the check inside is_product_supported that will make even more sense